### PR TITLE
Allow Deployment strategy customization when installing the Helm chart

### PR DESCRIPTION
**Pull request by wallrj**
In https://github.com/cert-manager/cert-manager/pull/1091 @yann-soubeyrand added the following feature to the cert-manager Helm chart:

> This PR modifies cert-manager chart to allow customization of [Deployment strategy](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy). This way one can modify maxSurge and maxUnavailable values.
> 
> This could be needed when deploying cert-manager on each master node and setting affinities to forbid two pods on the same node. In this situation, default values of maxSurge (25% round up to next integer = 1) and maxUnavailable (25% round down to next integer = 0) block cert-manager update as the new surge pod can't be scheduled on a master node due to the anti affinities. Setting maxSurge to 0 and maxUnavailable to 1 would solve the problem.

I've copied Yann's Helm chart feature to the approver-policy chart, 
1. in the interests of increasing consistency between the Helm charts of the cert-manager projects, and 
2. assuming that the same use-case also applies to the approver-policy Deployment,

Yann added that feature 6 years ago and I am unsure whether the additions of [PodDisruptionBudget ](https://github.com/cert-manager/cert-manager/pull/3931)and [TopologySpreadConstraint ](https://github.com/cert-manager/cert-manager/pull/5395) have obsoleted this use-case.

## Testing

```yaml
# values.yaml
strategy:
  type: RollingUpdate
  rollingUpdate:
    maxSurge: 0
    maxUnavailable: 1
```

```sh
$ helm template deploy/charts/approver-policy/ | grep -i strategy

$ helm template deploy/charts/approver-policy/ -f values.yaml | grep -i  -C4 strategy
  replicas: 1
  selector:
    matchLabels:
      app: cert-manager-approver-policy
  strategy:
    rollingUpdate:
      maxSurge: 0
      maxUnavailable: 1
    type: RollingUpdate


$ helm upgrade approver-policy deploy/charts/approver-policy/ --install --namespace cert-manager --create-namespace -f values.yaml
Release "approver-policy" does not exist. Installing it now.
NAME: approver-policy
LAST DEPLOYED: Tue Mar  5 13:17:19 2024
NAMESPACE: cert-manager
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
⚠️  WARNING: Consider increasing the Helm value `replicaCount` to 2 if you require high availability.
⚠️  WARNING: Consider setting the Helm value `podDisruptionBudget.enabled` to true if you require high availability.

CHART NAME: cert-manager-approver-policy
CHART VERSION: v0.0.0
APP VERSION: v0.0.0

cert-manager-approver-policy is a cert-manager project.

If you're a new user, we recommend that you read the [cert-manager Approval Policy documentation] to learn more.

[cert-manager Approval Policy documentation]: https://cert-manager.io/docs/policy/approval/


$ kubectl get deploy -n cert-manager -o yaml | grep -C 4 strategy
    revisionHistoryLimit: 10
    selector:
      matchLabels:
        app: cert-manager-approver-policy
    strategy:
      rollingUpdate:
        maxSurge: 0
        maxUnavailable: 1
      type: RollingUpdate
```

### CODE REVIEW

File: deploy/charts/approver-policy/README.md
Comment (wallrj) [inline deploy/charts/approver-policy/README.md:1]: https://github.com/cert-manager/approver-policy/blob/d9ca7693911bd4b399370822afc7acc060f8265a/deploy/charts/approver-policy/README.md#strategy--object
image

File: deploy/charts/approver-policy/values.yaml
Comment (wallrj) [inline deploy/charts/approver-policy/values.yaml:272]: * https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy

**Comment by wallrj** [comment]
/hold 

I'm interested to learn whether @yann-soubeyrand  still uses this feature in cert-manager, or whether PodDisruptionBudger and TopologySpreadConstraints have made this obsolete.
It would be a shame to unnecessarily complicate the Helm chart if this feature is no longer useful.

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve

My belief is that PDBs and topologySpreadConstraints don't obsolete this but I leave the hold pending feedback on that! Thanks for this Richard!

**Comment by wallrj** [comment]
/unhold

Ok. Merging now. We can always deprecate the flag if it turns out not to be useful.

**Pull request merged**
